src/component/utils/exploration/__tests__/__snapshots__/wrapBlockNodes.js.snap

```diff
@@ -0,0 +1,799 @@
+<div>
+  <ol
+    className="public__DraftStyleDefault__ol"
+    data-offset-key="C-0-0"
+  >
+    <div
+      className="public__DraftStyleDefault__block"
+      data-offset-key="C-0-0"
+    >
+      <span
+        data-offset-key="C-0-0"
+        style={Object {}}
+      >
+        <span
+          data-text="true"
+        >
+          deeply nested list
+        </span>
+      </span>
+    </div>
+  </ol>
+</div>
+`;
+
+exports[`renders block with nested child that have same wrapperElement 1`] = `
+<div>
+  <ul
+    className="public__DraftStyleDefault__ul"
+    data-offset-key="C-0-0"
+  >
+    <div
+      className="public__DraftStyleDefault__block"
+      data-offset-key="C-0-0"
+    >
+      <span
+        data-offset-key="C-0-0"
+        style={Object {}}
+      >
+        <span
+          data-text="true"
+        >
+          deeply nested list
+        </span>
+      </span>
+    </div>
+  </ul>
+</div>
+`;
+
+exports[`renders block with nested child that is of different block type 1`] = `
+<div>
+  <ol
+    className="public__DraftStyleDefault__ol"
+    data-offset-key="A-0-0"
+  >
+    <blockquote
+      data-block={true}
+      data-editor="editor"
+      data-offset-key="A-0-0"
+    >
+      <div
+        className="public__DraftStyleDefault__block"
+        data-offset-key="A-0-0"
+      >
+        <ol
+          className="public__DraftStyleDefault__ol"
+          data-offset-key="B-0-0"
+        >
+          <li
+            data-block={true}
+            data-editor="editor"
+            data-offset-key="B-0-0"
+          >
+            <div
+              className="public__DraftStyleDefault__block"
+              data-offset-key="B-0-0"
+            >
+              <h1
+                data-block={true}
+                data-editor="editor"
+                data-offset-key="C-0-0"
+              >
+                <div
+                  className="public__DraftStyleDefault__block"
+                  data-offset-key="C-0-0"
+                >
+                  <span
+                    data-offset-key="C-0-0"
+                    style={Object {}}
+                  >
+                    <span
+                      data-text="true"
+                    >
+                      header inside list
+                    </span>
+                  </span>
+                </div>
+              </h1>
+            </div>
+          </li>
+        </ol>
+      </div>
+    </blockquote>
+    <div
+      className="public__DraftStyleDefault__block"
+      data-offset-key="B-0-0"
+    >
+      <h1
+        data-block={true}
+        data-editor="editor"
+        data-offset-key="C-0-0"
+      >
+        <div
+          className="public__DraftStyleDefault__block"
+          data-offset-key="C-0-0"
+        >
+          <span
+            data-offset-key="C-0-0"
+            style={Object {}}
+          >
+            <span
+              data-text="true"
+            >
+              header inside list
```

Comment:
Again here the 'header inside list' item shows up twice in the output.

This is really interesting. The code and tests all make sense - I'll continue thinking about this and see if I can come up with theories about why the output isn't right.